WikiWatershed/model-my-watershed
# MapShed Compare: Fix Line Chart Issue

## Overview

As seen in the Compare View GWLF-E Hydrology Charts, when there are multiple scenarios with identical values such that the lines overlap exactly, we see errors in the console. This is due to an underlying assumption made by the d3-voronoi library that all the vertices are distinct. When they are not, we see these errors. I found a PR https://github.com/novus/nvd3/pull/584 that removed duplicate vertices and adopted that into our shim. The charts still work as they should, and we no longer see the issues in the console.

Also made https://github.com/azavea/nvd3/pull/3 for recording this in our shim.

Connects #2928 

### Notes

I also tried disabling voronoi completely with `chart.useVoronoi(false)`, but that disabled the tooltips, which we do want to keep. More investigative details can be found in https://github.com/WikiWatershed/model-my-watershed/issues/2928#issuecomment-415165773.

## Testing Instructions

* Check out this branch and `bundle --vendor`
* Go to a GWLF-E project and open the compare view. Ensure there are no errors in the console.
* Switch between Hydrology and Water Quality. Switch between Charts and Table. Ensure there are no console errors.
